src/Components/Onboarding/Steps/Artists/Mutations/UpdateArtistFollow.ts

```diff
@@ -0,0 +1,32 @@
+import * as Relay from "react-relay"
+import { State } from "../ItemLink"
+
+type RelayMutationProps = State
+
+export default class UpdateArtistFollowMutation extends Relay.Mutation<RelayMutationProps, any> {
+  getMutation() {
+    return Relay.QL`mutation {
+      followArtist
+    }`
+  }
+
+  getVariables() {
+    return {
+      artist_id: this.props.artist_id,
+    }
+  }
+
+  getFatQuery() {
+    return Relay.QL`
+    fragment on Artist {
+      is_followed,
+      counts {
+        follows
+      }
+    }`
+  }
```

Comment:
This is where you would indicate that in response to your mutation you _also_ want to fetch an artist related to the mutated artist. Imagine that MP’s schema exposed a connection of artists related to an artist (which it does not have atm afaik):

```graphql
fragment on Artist {
  related_artists(first: 1) {
    name
  }
}
```

Reply:
So guess what :D, it's already there (in the schema at least, not necessarily within the mutation code?)

if you load up GraphiQL, and use this query (requires auth headers):
```
mutation {
  followArtist(input: {
    artist_id: "damien-hirst",
  }) {
    artist {
      id
      artists(size: 1) {
        name
      }
    }	
  }
}
```

You get a response that has the top related artist for this artist. What @xtina-starr and I were wondering though, is how an `ItemLink` would replace itself... because upon success / completion, we'd want to ditch all the props that were there (from the _followed_ artist), and replace them with props from the related artist... So setting the related artist field as it were wouldn't be useful right? because actually, that should become _the_ artist now...?

Reply:
@alloy we also noticed that we wanted to get an artist through the `/me/suggested/artists` endpoint and trying to figure out how to get there. It seems the Artist is returned and we would want access to the me schema.

Reply:
for the second point, we could just add `suggested_artists` as a connection to the `Artist` schema. Since that already exists nested under ` me `, I think it probably makes the most sense to include it in the `Artist` schema. Having it under `me` is just a relic (I'm pretty sure), of the old-school thinking where we were trying to map Metaphysics to Gravity.

It would also give us a chance to create a new loader for it, and also replace the legacy loader used in the `me` one.